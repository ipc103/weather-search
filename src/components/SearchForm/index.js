import React from 'react'

const SearchForm = ({city, state, handleChange, handleSubmit}) => {
  return (
    <form onSubmit={handleSubmit} >
      <div className='form-group'>
        <label>City</label>
        <input type='text' name='city' value={city} onChange={handleChange} className='form-control'/>
      </div>
      <div className='form-group'>
        <label>State</label>
        <input type='text' name='st' value={state} onChange={handleChange} className='form-control'/>
      </div>
      <input type='submit' value="Search" />
    </form>
  )
}

export default SearchForm
