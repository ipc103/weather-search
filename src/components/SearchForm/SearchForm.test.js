import React from 'react'
import {shallow} from 'enzyme'
import SearchForm from './'

describe('SearchForm', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<SearchForm />)
    expect(wrapper).toMatchSnapshot()
  });
})
