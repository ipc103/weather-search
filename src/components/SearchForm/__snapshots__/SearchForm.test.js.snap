// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <form
    onSubmit={undefined}
>
    <div
        className="form-group"
    >
        <label>
            City
        </label>
        <input
            className="form-control"
            name="city"
            onChange={undefined}
            type="text"
            value={undefined}
        />
    </div>
    <div
        className="form-group"
    >
        <label>
            State
        </label>
        <input
            className="form-control"
            name="st"
            onChange={undefined}
            type="text"
            value={undefined}
        />
    </div>
    <input
        type="submit"
        value="Search"
    />
</form>,
  "nodes": Array [
    <form
      onSubmit={undefined}
>
      <div
            className="form-group"
      >
            <label>
                  City
            </label>
            <input
                  className="form-control"
                  name="city"
                  onChange={undefined}
                  type="text"
                  value={undefined}
            />
      </div>
      <div
            className="form-group"
      >
            <label>
                  State
            </label>
            <input
                  className="form-control"
                  name="st"
                  onChange={undefined}
                  type="text"
                  value={undefined}
            />
      </div>
      <input
            type="submit"
            value="Search"
      />
</form>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <form
          onSubmit={undefined}
>
          <div
                    className="form-group"
          >
                    <label>
                              City
                    </label>
                    <input
                              className="form-control"
                              name="city"
                              onChange={undefined}
                              type="text"
                              value={undefined}
                    />
          </div>
          <div
                    className="form-group"
          >
                    <label>
                              State
                    </label>
                    <input
                              className="form-control"
                              name="st"
                              onChange={undefined}
                              type="text"
                              value={undefined}
                    />
          </div>
          <input
                    type="submit"
                    value="Search"
          />
</form>,
        "_debugID": 2,
        "_renderedOutput": <form
          onSubmit={undefined}
>
          <div
                    className="form-group"
          >
                    <label>
                              City
                    </label>
                    <input
                              className="form-control"
                              name="city"
                              onChange={undefined}
                              type="text"
                              value={undefined}
                    />
          </div>
          <div
                    className="form-group"
          >
                    <label>
                              State
                    </label>
                    <input
                              className="form-control"
                              name="st"
                              onChange={undefined}
                              type="text"
                              value={undefined}
                    />
          </div>
          <input
                    type="submit"
                    value="Search"
          />
</form>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown />,
}
`;
